lib/puppet/pops/visitor.rb

```diff
@@ -24,11 +24,14 @@ def initialize(receiver, message, min_args=0, max_args=nil)
 
   # Visit the configured receiver
   def visit(thing, *args)
-    visit_this(@receiver, thing, *args)
+    visit_this(@receiver, thing, args)
   end
 
+  DOUBLE_COLON = '::'
+  NO_ARGS = []
```

Comment:
should be frozen since it is passed on in calls, and faulty code could add values to it (which would be very bad).